### DISCUSSION TITLE
It's possible to repeat an svg to use like a background?

### DISCUSSION THREAD

**Discussion by gabedesanta** [slint-ui/slint#3333] [Q&A]
Hi,

I would like to begin by saying that I am in the process of learning slint, and so far, I have been enjoying it. Therefore, I apologize if this question has already been answered before. _(Even though I haven't come across any.)_

I am creating an application that has a "2bit" design, and it uses this pattern from the photo below as the background for some components. This pattern could be recreated using an SVG, repeated multiple times until it reaches the component's size limit.

![pattern](https://github.com/slint-ui/slint/assets/143060187/f73ec335-9029-4a31-9507-ac74570589c3)
(The svg is a 4x4px with the top left and bottom right being a white pixel)

Is this possible to achieve? From what I've researched, it's not possible to have an image automatically repeating in this manner. As you have in CSS with "background-repeat" or in Swift `Image("background").resizable(resizingMode: .tile)`.

I tried using Image with HorizontalLayout and VerticalLayout inside a custom button component based on a Rectangle, but I either end up compromising performance (presumably due to numerous SVGs) or I encounter a recursion error, .

**Comment by tronical** [comment]
That's true, we don't have something like `background-repeat`. We could either make this a feature of the `Image` element perhaps, or we could add support for images to the `brush` type, along with a repetition enum. @ogoffart @FloVanGH what do you think?

Meanwhile, I think your approach of repeating manually with a layout should be fine. If this is slow, then you could try to use the [`cache-rendering-hint`](https://slint.dev/releases/1.1.1/docs/slint/src/builtins/elements#miscellaneous) on a `Rectangle` that _contains_ your layout. Does that help?

**Comment by ogoffart** [comment]
Something like [that](https://slintpad.com/?snippet=import+%7B+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Acomponent+RepeatedImage+%7B%0A++++preferred-height%3A+100%25%3B%0A++++preferred-width%3A+100%25%3B%0A++++in+property+%3Cimage%3E+source%3B%0A++++Rectangle+%7B%0A++++++++clip%3A+true%3B%0A++++++++HorizontalLayout+%7B%0A++++++++++++for+_+in+%28root.width%2F%28source.width%2B1%29%2F1px%29+%3A+VerticalLayout+%7B%0A++++++++++++++++for+_+in+%28root.height%2F%28source.height%2B1%29%2F1px%29+%3A+Image+%7B%0A++++++++++++++++++++source%3A+root.source%3B%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++height%3A+200px%3B%0A++++width%3A+300px%3B%0A++++RepeatedImage+%7B%0A++++++++source%3A+%40image-url%28%22https%3A%2F%2Fslint.dev%2Flogo%2Fslint-logo-circle-dark.svg%22%29%3B%0A++++%7D%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++Text+%7B%0A++++++++++++text%3A+%22Hello%5CnWorld%21%22%3B%0A++++++++++++font-size%3A+24px%3B%0A++++++++++++horizontal-alignment%3A+center%3B%0A++++++++++++color%3A+red%3B%0A++++++++%7D%0A++++++++%0A++++++++HorizontalLayout+%7B+alignment%3A+center%3B+Button+%7B+text%3A+%22OK%21%22%3B+%7D+%7D%0A++++%7D%0A%7D%0A)
